train_nsf_sim_cache_sid_load_pretrain.py

```diff
@@ -122,13 +105,13 @@ def run(rank, n_gpus, hps):
         collate_fn = TextAudioCollate()
     train_loader = DataLoader(
```

Comment:
we need to probably make sure that `train_dataset` is loading everything to GPU RAM, and serves samples directly from there

Reply:
How?
Also, in a followup PR perhaps?

Reply:
ok, we need a list of all the future fixes

Reply:
`train_dataset` is initialized with: `TextAudioLoaderMultiNSFsid(hps.data.training_files, hps.data)`, where:
`hps.data = {'max_wav_value': 32768.0, 'sampling_rate': 40000, 'filter_length': 2048, 'hop_length': 400, 'win_length': 2048, 'n_mel_channels': 125, 'mel_fmin': 0.0, 'mel_fmax': None, 'training_files': '/tmp/tmpqzg__17h/rvc/logs/5ae7c291-015c-4335-be31-4bde555765e1/filelist.txt'}`.

The `TextAudioLoaderMultiNSFsid` class is defined [here](https://github.com/LightricksResearch/rvc/blob/main/lib/train/data_utils.py#L10) (never touched it until now).
Don't see any mention of device/cuda there, so I doubt it uses GPU memory, but 🤷.

Reply:
not super simple.. 
we need to decide how important it is to fix